db/document.go

```diff
@@ -316,6 +316,22 @@ func (doc *Document) BodyBytes() ([]byte, error) {
 	return doc._rawBody, nil
 }
 
+func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, error) {
+	var userXattr interface{}
+	if len(doc.rawUserXattr) > 0 {
+		err := base.JSONUnmarshal(doc.rawUserXattr, &userXattr)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return map[string]interface{}{
```

Comment:
The updated unit tests are passing a nil value for metaMap to handle the case where userXattrKey isn't set, but this looks like it's going to return a map with an empty key and null value.  That doesn't seem desirable.  

I think an early return when userXattrKey is empty also makes sense.

Reply:
Yeah it does leave it as null. I can early exit. If we want to.